### ISSUE TITLE
abstract method with no parameters shows wrong info

### ISSUE THREAD

**Issue by myphysicslab** [jleyba/js-dossier#35]
The case shown in issue #30 has been fixed, but there are some other cases that are still wrong. They seem to all be when the method has ***no parameters***, but does have a return type.  For example, we get this:

```
getBottomBody(arg0)

Returns the bottom-most position of this Edge, in body coordinates.

Specified by: myphysicslab.lab.engine2D.Edge

Parameters
arg0...?
```

instead of:

```
getBottomBody()

Returns the bottom-most position of this Edge, in body coordinates.

Returns
number
bottom-most position of this Edge, in body coordinates
```

From the `Edge` interface:
```javascript
/** Returns the bottom-most position of this Edge, in body coordinates.
@return {number} bottom-most position of this Edge, in body coordinates
*/
Edge.prototype.getBottomBody;
```

From `AbstractEdge`
```javascript
/** @inheritDoc */
AbstractEdge.prototype.getBottomBody = goog.abstractMethod;
```

**Comment by jleyba** [comment]
Can you point me at your JS code?

I'm having some trouble reproducing this in a test. I defined `Edge` as:

```
/** @interface */
var Edge = function() {};
/** @return {number} x. */
Edge.prototype.id;
```

Which results in `id` being recorded as a _property_ with unknown type, not a function.

**Comment by myphysicslab** [comment]
Strange, it's now working perfectly fine.  These are showing up with the return type and in the Instance Methods section as expected.

Here is an excerpt of the code:
```javascript
goog.provide('myphysicslab.lab.engine2D.Edge');

goog.require('myphysicslab.lab.engine2D.Vertex');

goog.scope(function() {

/** An Edge of a Polygon has a start and finish Vertex and belongs to a particular
Polygon. 
* @interface
*/
myphysicslab.lab.engine2D.Edge = function() {};
var Edge = myphysicslab.lab.engine2D.Edge;

/** Returns the bottom-most position of this Edge, in body coordinates.
@return {number} bottom-most position of this Edge, in body coordinates
*/
Edge.prototype.getBottomBody;

}); // goog.scope
```

The documentation produced is:
```
getBottomBody()

Returns the bottom-most position of this Edge, in body coordinates.

Returns
number
bottom-most position of this Edge, in body coordinates
```

**Comment by jleyba** [comment]
There are some actual bugs in this area.

```
/** @override */
AbstractEdge.prototype.getBottomBody = goog.abstractMethod;
```

If this shows up as an unknown type on `AbstractEdge`, we should pull the type from the interface (I still don't understand exactly why the compiler doesn't resolve it immediately).

I'm going to leave this open as a tracking bug.

**Comment by myphysicslab** [comment]
Oops, I was looking in the wrong place: the bug is still there in AbstractEdge.  The docs are OK for Edge interface and for the concrete classes StraightEdge and CircularEdge.  The latter two have in their constructors:

```
@extends {myphysicslab.lab.engine2D.AbstractEdge}
```

and they of course have actual implementations of the abstract methods, for example:

```javascript
/** @inheritDoc */
StraightEdge.prototype.getBottomBody = function() {
  var y1 = this.v1_.locBodyY();
  var y2 = this.v2_.locBodyY();
  return y1 < y2 ? y1 : y2;
};
```

**Comment by myphysicslab** [comment]
Problem still occurs in 6dfa54799f485d58ee65309cc7feabcc6726adfe

There are 7 methods in AbstractEdge that have this problem; what they have in common is:

1. are assigned `goog.abstractMethod`
2. have no parameter
3. return a simple type such as number, boolean, or undefined.

There are 13 other methods that are assigned `goog.abstractMethod` which don't have the problem, and they all either have a parameter or return value that is a "complex type" (some other object).

I've put the source code for `Edge.js` and `AbstractEdge.js` up on my website at
+ <http://www.myphysicslab.com/rmnp/AbstractEdge.js>
+ <http://www.myphysicslab.com/rmnp/Edge.js>

Those won't compile without a lot of other code... I'll have it all up on github before long, I could do that sooner if it's useful.

The resulting dossier documentation is at
+ <http://www.myphysicslab.com/rmnp/class_myphysicslab_lab_engine2D_AbstractEdge.html>
+ <http://www.myphysicslab.com/rmnp/interface_myphysicslab_lab_engine2D_Edge.html>

**Comment by myphysicslab** [comment]
Confirmed fixed in 0121f527423cde85cc8d05c1663b505db93ea321
